song/song_.py

```diff
@@ -75,13 +75,13 @@ def _edge_curation(self) -> None:
         In this implementation, the graph always keep symmetric.
         """
         i_1 = self.neighbor_idxs[0]
-        for j in self.topology[i_1].keys():
+        for j in list(self.topology[i_1].keys()):
```

Comment:
When an element is deleted in the FOR loop, RuntimeError raised. So  covering by the list()